explorer/interpreter/action_stack.h

```diff
@@ -118,15 +118,29 @@ class ActionStack {
   // Suspends execution of the currently-executing continuation.
   auto Suspend() -> ErrorOr<Success>;
 
+  void Pop() { todo_.Pop(); }
+
  private:
   // Pop any ScopeActions from the top of the stack, propagating results as
   // needed, to restore the invariant that todo_.Top() is not a ScopeAction.
-  void PopScopes();
+  auto PopScopes() -> std::list<std::unique_ptr<Action>>;
 
   // Set `result` as the result of the Action most recently removed from the
   // stack.
   void SetResult(Nonnull<const Value*> result);
 
+  auto UnwindToWithCaptureScopesToDestroy(Nonnull<const Statement*> ast_node)
+      -> std::list<std::unique_ptr<Action>>;
+
+  auto UnwindPastWithCaptureScopesToDestroy(Nonnull<const Statement*> ast_node)
+      -> std::list<std::unique_ptr<Action>>;
+
+  void DestroyAllScopes(std::list<std::unique_ptr<Action>>&& actions);
+
+  void DestroyScopes(std::list<std::unique_ptr<Action>>&& actions);
```

Comment:
What's the difference between these two methods? Do we need both of them?

Reply:
Yes! I have documented why!

Reply:
Hmm. I don't think that should be necessary; can you add a comment like "TODO: see if it's possible to combine these"?

Reply:
Done!

Reply:
Done!!!